perception/sub8_perception/nodes/colored_rectangle_finder.py

```diff
@@ -0,0 +1,609 @@
+            epsilon_range=self.epsilon_range,
+            epsilon_step=self.epsilon_step)
+        if corners is None:
+            return False, self.magenta
+        self.last2d = self.rect_model.get_pose_2D(corners)
+        self.last_found_time_2D = self.image_sub.last_image_time
+        if self.do_3D:
+            if self.internal_active:
+                if not self._get_pose_3D(corners):
+                    return False, self.cyan
+
+        return True, self.green
+
+    def _get_edges(self):
+        '''
+        Proccesses latest image to find edges by:
+        blurring and thresholding for highly saturated orangish objects
+        then runs canny on threshold images and returns canny's edges
+        '''
+        blur = cv2.blur(self.last_image, (5, 5))
+        if self.color_space == 'BGR':
+            colored = blur
+        if self.color_space == 'HSV':
+            colored = cv2.cvtColor(blur, cv2.COLOR_BGR2HSV)
+        if self.color_space == 'LAB':
+            colored = cv2.cvtColor(blur, cv2.COLOR_BGR2LAB)
+
+        kernel = np.ones((3, 3), np.uint8)
+        erosion = cv2.erode(colored, kernel, iterations=6)
+
+        self.thresh_no_erosion = cv2.inRange(
+            colored, self.thresh_low, self.thresh_high)
+        thresh = cv2.inRange(erosion, self.thresh_low, self.thresh_high)
+
+        return thresh
+
+    def minimum_norm(self, center, features, means, channel=0):
+        '''
+        Among the provided features find the proximity to a predefined center
+
+        Center of feature may be geometric (ie. the 2D coordinates of the center of a contour),
+        or it may be proximity of a color to another
+
+        @param center   : base value to compare set of means against
+        @param features : the set of features who
+        @param means    : the corresponding set means of given set of features
+        @param channel  : channel to check along feature
+        '''
+        distances = []
+        for mean in means:
+            distances.append(np.linalg.norm(center[channel:] - mean[channel:]))
+        min_idx = distances.index(min(distances))
+        return features[min_idx]
+
+    def center_contour(self, contour):
+        ''' Find the geometric center of a provided contour. '''
+        center = cv2.moments(contour)
+        cX = int(center["m10"] / center["m00"])
+        cY = int(center["m01"] / center["m00"])
+        return np.array([cX, cY])
+
+    def image_mux_today(self, images):
+        ''' Plotting tool for debugging'''
+        labels = ['Original', 'Threshed', 'Full Mask', None]
+        # size = (960, 1280)
+        size = (480, 640)
+        t = ImageMux(
+            size=size, border_color=(
+                0, 0, 255), border_thickness=1, shape=(
+                2, 2), labels=labels, text_scale=1, text_color=(
+                255, 255, 0))
+        for i in xrange(len(images)):
+            t[i] = np.array(images[i])
+        cv2.imshow('Kapu', t.image)
```

Comment:
It still does. This is for the debug gui, otherwise it needs to be kind of big in rviz/rqt when running, you still think it should be the same for both?